lib/internal/http2/core.js

```diff
@@ -2979,6 +2988,7 @@ class Http2SecureServer extends TLSServer {
     if (typeof requestListener === 'function')
       this.on('request', requestListener);
     this.on('tlsClientError', onErrorSecureServerSession);
+    this.updateSettings = updateHttp2ServerSettings.bind(this);
```

Comment:
I'd prefer to just have this as a method in a class explicitly without bind, I think 3 line code-duplication is fine in this case, wdyt?
```js
updateSettings(settings) {
  assertIsObject(settings, 'settings');
  validateSettings(settings);
  this[kOptions].settings = { ...this[kOptions].settings, ...settings };
}
```

Also, perhaps use object spread in this case ` this[kOptions].settings = { ...this[kOptions].settings, ...settings };`?

Reply:
1. > I'd prefer to just have this as a method in a class explicitly without bind, I think 3 line code-duplication is fine in this case, wdyt?
2. > Also, perhaps use object spread in this case ` this[kOptions].settings = { ...this[kOptions].settings, ...settings };`?

I do not totally agree with your first point since it might be cumbersome to update both functions later. Still, I see that it is how `setTimeout` was implemented. I will do the same for updateSettings.

I agree with your second point. I will use object spread instead of `ObjectAssign`.